Oscil12.h

```diff
@@ -0,0 +1,325 @@
+/*
```

Comment:
Higher bit oscillators make a lot of sense on the 32 bitters, but please split them out of _this_ PR, as they are only partially related.

(And I think we'll probably want a more generic - templated based - solution, then.)

Reply:
Well they make sense on AVR with Timer4 (Leonardo, Lilypad USB) with 10-bit (and maybe 11) PWM. Because they are in a separate header file there is no cost to include them.  I am not sure how to templatize this well as you have to change the PROGRAM call according to byte or word lookup.

Reply:
I just confirmed that the FM example runs fine with a 16-bit carrier table on Lilypad USB (8MHz) and Teensy 2.0

Reply:
To clarify, I am absolutely _not_ opposed to a change along those lines, just please make it a separate pull request.

As for the templating, I think the solution will be to templatize CONSTABLE_READ(), too. But again, separate PR, please.

(And while doing this, it _may_ - or may not - make sense to also take a look whether Sample and Oscil can be made to share a common template.)